contracts/src/L1/rollup/ScrollChain.sol

```diff
@@ -527,19 +623,42 @@ contract ScrollChain is OwnableUpgradeable, PausableUpgradeable, IScrollChain {
             }
         }
 
+        // stack too deep
+        {
+            uint64 lastAppliedL1BlockInChunk = ChunkCodec.lastAppliedL1BlockInChunk(l2TxPtr);
+            chunkResult._l1BlockRangeHashInChunk = ChunkCodec.l1BlockRangeHashInChunk(l2TxPtr);
+
+            require(
+                lastAppliedL1BlockInChunk == chunkResult._lastAppliedL1BlockInChunk,
+                "incorrect lastAppliedL1Block in chunk"
+            );
+        }
+
         // check the actual number of transactions in the chunk
         require((dataPtr - txHashStartDataPtr) / 32 <= maxNumTxInChunk, "too many txs in one chunk");
 
-        // check chunk has correct length
-        require(l2TxPtr - chunkPtr == _chunk.length, "incomplete l2 transaction data");
+        // check chunk has correct length.
+        // 40 is the size of lastAppliedL1Block and l1BlockRangeHash.
+        require(l2TxPtr - chunkPtr + 40 == _chunk.length, "incomplete l2 transaction data");
+
+        // stack too deep
+        {
+            uint256 _lastAppliedL1BlockInChunk = chunkResult._lastAppliedL1BlockInChunk;
+            bytes32 _l1BlockRangeHashInChunk = chunkResult._l1BlockRangeHashInChunk;
+            assembly {
+                mstore(dataPtr, _lastAppliedL1BlockInChunk)
+                mstore(dataPtr, _l1BlockRangeHashInChunk)
+                dataPtr := add(dataPtr, 0x28)
```

Comment:
There is a check `require((dataPtr - txHashStartDataPtr) / 32 <= maxNumTxInChunk, "too many txs in one chunk");` a few lines above. Should this be done before the check?